test.py

```diff
@@ -0,0 +1,147 @@
+"""
+Compare the output of the current Securify with past outputs.
+
+Author: Quentin Hibon
+
+Copyright 2018 ChainSecurity AG
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+"""
+import json
+import shutil
+import subprocess
+import tempfile
+
+from pathlib import Path
+
+import psutil
+
+
+class MismatchError(Exception):
+    """Raised when there is a mismatch between past and current output
+    """
+    pass
+
+
+def raise_mismatch(expected, current):
+    """Raises the formatted exception.
+    """
+    raise MismatchError(f'Current: {current}, '
+                        f'expected: {expected}')
+
+
+def check_every_match(curr_res, exp_res, ctr, pat,
+                      keys=('violations', 'warnings', 'safe', 'conflicts')):
+    """Checks every match of a given pattern.
+    """
+    for k in keys:
+        c_pat_results, e_pat_results = curr_res[pat][k], exp_res[pat][k]
+        if c_pat_results != e_pat_results:
+            print(f'Different output in {ctr} '
+                  f'with pattern {pat} with {k}!')
+            raise_mismatch(e_pat_results, c_pat_results)
+
+
+def check_all_patterns(curr_json, expc_json, ctr):
+    """Checks every pattern from the given json
+    """
+    res = 'results'
+    curr_res, exp_res = curr_json[ctr][res], expc_json[ctr][res]
+
```

Comment:
We need to check that a pattern is not missing in the new output. If one pattern was omitted in the new output (e.g. because it was accidentally commented out) we will get no warning here at all.